src/clustering/administration/main/serve.hpp

```diff
@@ -85,6 +87,13 @@ struct service_address_ports_t {
     int port_offset;
 };
 
+struct tls_configs_t {
+    tls_configs_t() : web(NULL), driver(NULL), cluster(NULL) {}
```

Comment:
After thinking about it a little, I think the easiest way for making `SSL_CTX` RAII is by using an `std::shared_ptr`. This is how you can do it:
```C++
#include <memory>
...

typedef std::shared_ptr<SSL_CTX> shared_ssl_ctx_t;
```

Then in your `initialize_tls_ctx` function, the first line would become:
+    *tls_ctx = SSL_CTX_new(SSLv23_method())`